docs/posts/1brc/index.qmd

```diff
@@ -0,0 +1,343 @@
+---
+title: "Using one Python dataframe API to take the billion row challenge with DuckDB, Polars, and DataFusion"
+author: "Cody"
+date: "2024-01-22"
+categories:
+    - blog
+    - duckdb
+    - polars
+    - datafusion
+---
+
+## Overview
+
+This is an implementation of the [The One Billion Row
+Challenge](https://www.morling.dev/blog/one-billion-row-challenge/):
+
+> Let’s kick off 2024 true coder style—​I’m excited to announce the One Billion
+> Row Challenge (1BRC), running from Jan 1 until Jan 31.
+
+> Your mission, should you decide to accept it, is deceptively simple: write a
+> Java program for retrieving temperature measurement values from a text file and
+> calculating the min, mean, and max temperature per weather station. There’s just
+> one caveat: the file has 1,000,000,000 rows!
+
+I haven't written Java since dropping a computer science course my second year
+of college that forced us to do functional programming exclusively in Java.
+However, I'll gladly take the challenge in Python using Ibis! In fact, I did
+something like this (generating a billion rows with 26 columns of random numbers
+and doing basic aggregations) to test out DuckDB and Polars.
+
+In this blog, we'll demonstrate how Ibis provides a single Python dataframe API
+to take the billion row challenge with DuckDB, Polars, and DataFusion.
```

Comment:
```suggestion
In this blog, we'll demonstrate how to use Ibis's unified Python dataframe API
to complete the billion row challenge with DuckDB, Polars, and DataFusion.
```
"take" sounded like the wrong word to me, I think? Just a suggestion.

Reply:
gonna leave as "taking the challenge" and have a clickbaity linkedin post